packages/state-manager/src/sagas/communities/joinNetwork/joinNetwork.saga.ts

```diff
@@ -0,0 +1,47 @@
+import { CommunityOwnership, CreateNetworkPayload, InvitationDataVersion, SocketActionTypes } from '@quiet/types'
+import { PayloadAction } from '@reduxjs/toolkit'
+import { apply, put } from 'typed-redux-saga'
+import { Socket, applyEmitParams } from '../../../types'
+import { communitiesActions } from '../communities.slice'
+
+export function* joinNetworkSaga(
+  socket: Socket,
+  action: PayloadAction<ReturnType<typeof communitiesActions.joinNetwork>['payload']>
+) {
+  console.log('join network saga', action.payload)
+  const data = action.payload
+  let payload: CreateNetworkPayload
+
+  data.version = data.version || InvitationDataVersion.v1
+  switch (data.version) {
+    case InvitationDataVersion.v1:
+      console.log('join network saga invitation data v1')
+      payload = {
+        ownership: CommunityOwnership.User,
+        peers: data.pairs,
+        psk: data.psk,
+        ownerOrbitDbIdentity: data.ownerOrbitDbIdentity,
+      }
+      break
+    case InvitationDataVersion.v2:
+      console.log('join network saga invitation data v2')
+      const response: CreateNetworkPayload = yield* apply(
+        socket,
+        socket.emitWithAck,
+        applyEmitParams(SocketActionTypes.DOWNLOAD_INVITE_DATA, {
+          serverAddress: data.serverAddress,
+          cid: data.cid,
+        })
+      )
+      payload = {
+        ownership: CommunityOwnership.User,
+        peers: response.peers,
+        psk: response.psk,
+        ownerOrbitDbIdentity: response.ownerOrbitDbIdentity,
+      }
+      break
+  }
+
+  console.log('join network saga payload', payload)
+  yield* put(communitiesActions.createNetwork(payload))
+}
```

Comment:
You may be right. I added new saga because new flow of joining community requires some additional steps (that would be done as part of the next task). So right now it seems like I'm just creating payload object for createNetwork but I separated it from createNetwork with next steps in mind.

Reply:
I think most of the joining is going to occur in `launchCommunity`, while `createNetwork` is still just creating a Tor site.

Reply:
Resigning from joinNetwork saga is also fine by me if it would fit better with your refactoring.